cms/utils/__init__.py

```diff
@@ -16,15 +21,42 @@ def get_current_site():
     return Site.objects.get_current()
 
 
+@lru_cache(512)
+def get_object_language(request):
```

Comment:
Maybe get_language_from_content_object, not that important.

Please add a doc type to help anyone using htis in the core, as it's a util it is only supposed to be used in the core checking that is as intended. Helpers are for external reuse to be kept backwards compatible where possible.